### PR TITLE
fix(providers/api/metrics): Report correct error codes in metrics for APIs

### PR DESCRIPTION
**Pull request by g3ndo9kar9**

### CODE REVIEW

File: providers/apis/defi/raydium/types.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ var DefaultAPIConfig = oracleconfig.APIConfig{
 	Enabled:          true,
 	Name:             Name,
 	Timeout:          500 * time.Millisecond,
-	Interval:         100 * time.Millisecond,
+	Interval:         500 * time.Millisecond,
```
Comment (aljo242) [inline providers/apis/defi/raydium/types.go:83]: why this change?

Comment (g3ndo9kar9) [reply providers/apis/defi/raydium/types.go:83]: Based on our current RPC providers, we'll get rate-limited otherwise

File: providers/base/api/metrics/api_query_handler.go [hunk 2]
```diff
@@ -26,7 +27,7 @@ const (
 type APIMetrics interface {
 	// AddProviderResponse increments the number of ticks with a fully successful provider update.
 	// This increments the number of responses by provider, id (i.e. currency pair), and status.
-	AddProviderResponse(providerName, id string, status Status)
+	AddProviderResponse(providerName, id string, errorCode providertypes.ErrorCode)
```
Comment (aljo242) [inline providers/base/api/metrics/api_query_handler.go:30]: do we use the `Status` type anywhere now? 

is it safe / prudent to remove?

Comment (g3ndo9kar9) [reply providers/base/api/metrics/api_query_handler.go:30]: yep!

**Review by aljo242 — APPROVED** [review]

**Review by Eric-Warehime — APPROVED** [review]
